lib/lz4.h

```diff
@@ -466,10 +478,7 @@ LZ4_DEPRECATED("use LZ4_decompress_fast() instead") LZ4LIB_API int LZ4_uncompres
 LZ4_DEPRECATED("use LZ4_decompress_safe() instead") LZ4LIB_API int LZ4_uncompress_unknownOutputSize (const char* source, char* dest, int isize, int maxOutputSize);
 
 /* Obsolete streaming functions; use new streaming interface whenever possible */
-LZ4_DEPRECATED("use LZ4_createStream() instead") LZ4LIB_API void* LZ4_create (char* inputBuffer);
```

Comment:
Removing support for these functions could, in theory, force us to update the major number of library version. I would prefer not to.

Now, I understand that these functions, and more specifically `LZ4_slideInputBuffer()`, cannot work anymore, due to refurbishing of the pointer required for this action.

Also, this is a fairly complex (and badly specified) API, which never had a lot of clients, had a limited lifetime, was deprecated long ago, so the impact should be fairly limited by now.
Nevertheless, it doesn't take long to find [reference of these functions somewhere over Internet](https://www.google.com/search?q=LZ4_slideInputBuffer). It doesn't prove that they are still active, just that they are not forgotten.

As a consequence, I'm considering a mild intermediate scenario : 
Maybe these prototypes could continue to exist, for linking purpose (imagine that, for obscure reasons, these symbols are still linked within a deeply nested dependency that no-one triggers anyway), but always generate a documented error code if someone tries to use them ?

This is an opened question, feel free to comment.

Reply:
Yes, I see your point. I don't know if that would be an improvement necessarily.

An alternative is to get clever and pack the context better so we don't have to remove `bufferStart`. We never use the `uint8_t* dictionary` and `LZ4_stream_t_internal* dictCtx` fields at the same time, so they could be a `union` together, or replaced by a single `void*`.

Reply:
Hmm, no, that's not quite true. We only ever use one of them at a time, but there are contexts where we don't have any other information about which mode we're in, `usingExtDict` or `usingDictCtx`. So we would have to get pretty tricky. We could split `uint16_t tableType` into two `uint8_t`s, a `tableType` and `dictDirective`...

What do you think?

Reply:
Well, these functions are mostly dead.
They may come back haunting us as zombies, but we should not get out of our way to keep them alive.
Select a simple solution.
Even removing these functions feels better than introducing complex and risky mechanisms (that's still an opened possibility).
I initially liked the "return always error" one on the ground that it felt simple enough to implement, and reduce any potential interaction with the rest of the code.